airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/avro_parser.py

```diff
@@ -174,6 +174,10 @@ def _to_output_value(avro_format: AvroFormat, record_type: Mapping[str, Any], re
             return str(record_value)
         elif record_type.get("logicalType") == "date":
             return record_value.isoformat()
+        elif record_type.get("logicalType") == "timestamp-millis":
+            return record_value.isoformat(sep="T", timespec="milliseconds")
+        elif record_type.get("logicalType") == "timestamp-micros":
+            return record_value.isoformat(sep="T", timespec="microseconds")
```

Comment:
Change: previously we used the default serialization that pydantic used for these cases. However, pydantic v2 has strange behavior when there is a timezone, but the timezone is UTC ([issue](https://github.com/pydantic/pydantic/issues/7035#issuecomment-1706276095))

Therefore, we explicitly output all of our types with the `+00:00` format as expected in the `avro_all_types_stream` scenario.